File: MergeSort.java
import java.lang.System.*;
import java.util.*;

public class MergeSort {
  private static int size = 10;
  private static int[] arr = new int[]{2, 4, 6, 8, 10, 1, 3, 5, 7, 9};

  private static void printArray(int arr[], int size) {
    System.out.println("Main.");
    for(int i = 0; i < size; i++) {
      if(i == 0)
        System.out.print("{");
      else
        System.out.print(", ");
      System.out.print(arr[i]);
    }
    System.out.println("}");
  }

  public static void main(String[] args) {
    printArray(arr, size);
  }

}

File: MergeSort.cpp
#include <iostream>

using namespace std;

void printArray(int arr[], int n) {
	for(int i = 0; i < n; i++)
		cout << arr[i] << " ";
	cout << endl;
}

void merge(int* a, int low, int mid, int high) {
	int i = low, j = mid+1, k = 0;
	int b[high-low+1];

	while(i <= mid && j <= high)
		b[k++] = (a[i] < a[j]) ? a[i++] : a[j++];
	while(i <= mid)
		b[k++] = a[i++];
	while(j <= high)
		b[k++] = a[j++];

	for(i = 0, j = low; i < k; i++, j++)
		a[j] = b[i];
}

void mergesort(int* a, int low, int high) {
	if(low >= high)
		return;

	int mid = (high + low)/2;
	mergesort(a,low,mid);
	mergesort(a,mid+1,high);
	merge(a,low,mid,high);
}

int main(int argv, char** argc) {
	int n;
	cout << "How many elements in array?" << endl;
	cin >> n;

	int arr[n];
	
	cout << "Enter " << n << " elements:" << endl;
	for(int i = 0; i < n; i++) {
		cin >> arr[i];
	}

	printArray(arr,n);

	mergesort(arr,0,n-1);
	printArray(arr,n);
}
